v1.0/multi-active-availability.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Multi-Active Availability
+summary: Learn about CockroachDB's high availability model, known as Multi-Active Availability
+toc: false
+---
+
+CockroachDB's availability model is described as "Multi-Active Availability." In essence, multi-active availability provides benefits similar to traditional notions of high availability, but also lets you read and write from every node in your cluster without generating any conflicts.
+
+<div id="toc"></div>
+
+## What is High Availability?
+
+High availability lets an application continue running even if a system hosting one of its services fails. This is achieved by scaling the application's services horizontally, i.e., replicating the service across many machines or system. If any one of them fails, the others can simply step in and perform the same service.
+
+Before diving into the details of CockroachDB's multi-active availability, we'll review the two most common high availability designs: [Active-Passive](#active-passive) and [Active-Active](#active-active) systems.
+
+### Active-Passive
+
+In active-passive systems, all traffic is routed to a single, "active" replica. Changes to the replica's state are then copied to a backup "passive" replica, in an attempt to always mirror the active replica as closely as possible.
+
+However, this design has downsides:
+- If you use asynchronous replication, you cannot guarantee that any data is ever successfully replicated to passive followers––meaning you can easily lose data. Depending on your industry, this could have pretty dire consequences.
+- If you use synchronous replication and any passive replicas fail, you have to either sacrifice availability for the entire application or risk inconsistencies.
+
+### Active-Active
+
+In active-active systems, multiple replicas run identical services, and traffic is routed to all of them. If any replica fails, the others simply handle the traffic that would've been routed to it.
+
+For databases, though, active-active replication is incredibly difficult to instrument for most workloads. For example, if you let multiple replicas handle writes for the same keys, how do you keep them consistent?
+
+#### Example: Conflicts with Active-Active Replication
+
+For this example, we have 2 replicas (**A**, **B**) in an active-active high availability cluster.
+
+1. **A** receives a write for key `xyz` of `'123'`, and then immediately fails.
+2. **B** receives a read of key `xyz`, and returns a `NULL` because it cannot find the key.
+3. **B** then receives a write for key `xyz` of `'456'`.
+4. **A** is restarted and attempts to rejoin **B**––but what do you about key `xyz`? There's an inconsistency in the system without a clear way to resolve it.
+
+{{site.data.alerts.callout_info}}In this example, the cluster remained active the entire time. But in terms of the CAP theorem, this is an AP system; it favored being available instead of consistent.{{site.data.alerts.end}}
```

Comment:
Can / should we link to an explanation of CAP theorem? This is the first mention of it and I'd like to know more.

Also, does "AP" stand for "active-passive"?

Reply:
AP in the CAP theorem is "available/partition" which means that the system stays available (instead of consistent) when partitions occur. The other paradigm (which CockroachDB is part of) favors consistency over availability when partitions occur.

Reply:
Gotcha!